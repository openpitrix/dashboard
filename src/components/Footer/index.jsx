import React, { PureComponent } from 'react';
import { translate } from 'react-i18next';

import styles from './index.scss';

@translate()
export default class Footer extends PureComponent {
  getCurrentLocale = () => typeof window !== 'undefined' && localStorage.getItem('i18nextLng');

  changeLocale = (lang, e) => {
    e.preventDefault();
    if (this.getCurrentLocale() !== lang) {
      this.props.i18n.changeLanguage(lang);
    }
  };

  render() {
    const { t } = this.props;

    return (
      <div className={styles.footer}>
        <div className={styles.wrapper}>
          <div className={styles.inner}>
            <span className={styles.logo}>
              <img src="/logo_grey.svg" alt="logo" height="100%" />
            </span>
            <ul className={styles.terms}>
              <li>
                <a
                  target="_blank"
                  href="https://docs.openpitrix.io/home/"
                  rel="noopener noreferrer"
                >
                  {t('Document')}
                </a>
              </li>
              <li>
                <a
                  target="_blank"
                  href="https://openpitrix.io"
                  rel="noopener noreferrer"
                >
                  {t('About')}
                </a>
                {/* <span className={styles.dot} />
                 <a href="javascript:void(0)">{t('Help')}</a>
                 <span className={styles.dot} />
                 <a href="javascript:void(0)">{t('Terms')}</a> */}
              </li>
              <li>OpenPitrix &copy; 2018</li>
              <li>
                <a
                  href="#"
                  onClick={this.changeLocale.bind(null, 'zh')}
                  className={styles.toggleLocale}
                >
                  中文
                </a>
                <span className={styles.dot} />
                <a
                  href="#"
                  onClick={this.changeLocale.bind(null, 'en')}
                  className={styles.toggleLocale}
                >
                  English
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    );
  }
}
