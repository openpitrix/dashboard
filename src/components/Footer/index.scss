@import "~scss/vars";

$line-height: $footer-height - 1;
$fg-color: #78839e;

.footer {
  position: static;
  width: 100%;
  height: $footer-height;
  background-color: $N0;
  box-sizing: border-box;

  .wrapper {
    width: $content-width;
    margin: 0 auto;
    color: $fg-color;

    .logo {
      display: inline-block;
      float: left;
      font-size: $size-medium;
      line-height: $line-height;
      text-transform: uppercase;
      :global {
        img {
          height: 16px;
        }
      }
    }
    .terms {
      display:inline-block;
      float: right;
      line-height: $line-height;
      & > li{
        list-style-type: none;
        float: left;
        text-align: right;
        margin-left: 10px;

        &.copyright {
          margin-left: 20px;
        }

        a {
          color: $fg-color;
          &:hover {
            color: $P75;
          }
        }
      }
    }
  }

  .dot {
    display:inline-block;
    height: 3px;
    width: 3px;
    background-color: $fg-color;
    border-radius: 50%;
    vertical-align: middle;
  }

  .toggleLocale {
    &:hover {
      color: $P75 !important;
    }
  }
}
