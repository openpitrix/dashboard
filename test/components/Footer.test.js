import { BrowserRouter } from 'react-router-dom';

import Footer from 'components/Footer';

describe('Base/Footer', () => {
  it('basic render', () => {
    const wrapper = render(
        <BrowserRouter><Footer /></BrowserRouter>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
