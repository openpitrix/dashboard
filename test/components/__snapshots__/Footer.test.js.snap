// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base/Footer basic render 1`] = `
<div
  class="footer"
>
  <div
    class="wrapper"
  >
    <div
      class="inner"
    >
      <span
        class="logo"
      >
        <img
          alt="logo"
          height="100%"
          src="/logo_grey.svg"
        />
      </span>
      <ul
        class="terms"
      >
        <li>
          <a
            href="https://docs.openpitrix.io/home/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Document
          </a>
        </li>
        <li>
          <a
            href="/user/provider/apply"
          >
            Become App Provider
          </a>
        </li>
        <li>
          <a
            href="https://openpitrix.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            About
          </a>
        </li>
        <li>
          OpenPitrix Â© 2019
        </li>
      </ul>
    </div>
  </div>
</div>
`;
